ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -314,14 +292,66 @@ func convertToContainerDef(inputCfg *adapter.ContainerConfig, ecsContainerDef *C
 		outputContDef.LinuxParameters.SetTmpfs(inputCfg.Tmpfs)
 	}
 
-	// Set fields from ecs-params file
+	// initialize container resources from inputCfg
+	cpu := inputCfg.CPU
+	mem := inputCfg.Memory
+	memRes := inputCfg.MemoryReservation
+
+	// Set essential & resource fields from ecs-params file, if present
 	if ecsContainerDef != nil {
 		outputContDef.Essential = aws.Bool(ecsContainerDef.Essential)
+
+		cpu = getResourceValue(inputCfg.Name, cpu, ecsContainerDef.Cpu)
+
+		ecsMemInMB := adapter.ConvertToMemoryInMB(int64(ecsContainerDef.Memory))
+		mem = getResourceValue(inputCfg.Name, mem, ecsMemInMB)
+
+		ecsMemResInMB := adapter.ConvertToMemoryInMB(int64(ecsContainerDef.MemoryReservation))
+		memRes = getResourceValue(inputCfg.Name, memRes, ecsMemResInMB)
+	}
+
+	// One or both of memory and memoryReservation is required to register a task definition with ECS
+	// If neither is provided by 1) compose file or 2) ecs-params, set default
+	// NOTE: Docker does not set a memory limit for containers
+	if mem == 0 && memRes == 0 {
```

Comment:
I think this check needs to come after the `mem < memRes` check since otherwise `mem` will never be 0.

Eg. this test fails:
```
func TestConvertToTaskDefinitionWithECSParams_DefaultMemoryLessThanMemoryRes(t *testing.T) {
	// set up containerConfig w/o value for Memory
	containerConfig := &adapter.ContainerConfig{
		Name:              "web",
		Image:             "httpd",
		CPU:               int64(5),
	}

	// define ecs-params value we expect to be present in final containerDefinition
	ecsParamsString := `version: 1
task_definition:
  services:
    web:
      mem_reservation: 1g`

	content := []byte(ecsParamsString)

	tmpfile, err := ioutil.TempFile("", "ecs-params")
	assert.NoError(t, err, "Could not create ecs params tempfile")

	defer os.Remove(tmpfile.Name())

	_, err = tmpfile.Write(content)
	assert.NoError(t, err, "Could not write data to ecs params tempfile")

	err = tmpfile.Close()
	assert.NoError(t, err, "Could not close tempfile")

	ecsParamsFileName := tmpfile.Name()
	ecsParams, err := ReadECSParams(ecsParamsFileName)
	assert.NoError(t, err, "Could not read ECS Params file")

	containerConfigs := []adapter.ContainerConfig{*containerConfig}
	_, err = convertToTaskDefWithEcsParamsInTest(t, containerConfigs, "", ecsParams)

	assert.Error(t, err, "Expected error because memory reservation is greater than memory limit")
}
```

Reply:
added test, changed validation order (also changed mem < memRes check to work even if one is 0 (new test failed w/o it))